src/services/AlterUserService.ts

```diff
@@ -0,0 +1,93 @@
+import { getCustomRepository } from 'typeorm';
+
+import { verifyEmail, verifyUrl, verifyField } from '../utils/regex';
+import User from '../entities/User';
+import UsersRepositories from '../repositories/UsersRepositories';
+
+interface IAlterUserService {
+  username: string;
+  newName?: string;
+  newLastName?: string;
+  newProfileImageUrl?: string;
+  newBio?: string;
+  newEmail?: string;
+  newGender?: string;
+}
+
+class AlterUserService {
+  async execute({
+    username,
+    newName,
+    newLastName,
+    newProfileImageUrl,
+    newBio,
+    newEmail,
+    newGender,
+  }: IAlterUserService): Promise<User> {
+    const usersRepository = getCustomRepository(UsersRepositories);
+    const user = await usersRepository.findOne({ username });
+
+    if (user) {
+      let { name, lastName, profileImageUrl, bio, email, gender } = user;
+
+      if (newName) {
+        if (!verifyField(newName)) {
+          throw new Error('Invalid name');
+        }
+        name = newName;
+      }
+      if (newLastName) {
+        if (!verifyField(newLastName)) {
+          throw new Error('Invalid last name');
+        }
+        lastName = newLastName;
+      }
+      if (newProfileImageUrl) {
+        if (!verifyUrl(newProfileImageUrl)) {
+          throw new Error('Invalid profile image url');
+        }
+        profileImageUrl = newProfileImageUrl;
+      }
+      if (newBio) {
+        if (!verifyField(newBio)) {
+          throw new Error('Invalid bio');
+        }
+        bio = newBio;
+      }
+      if (newEmail) {
+        const emailAlreadyExists = await usersRepository.findOne({
+          email: newEmail,
+        });
+        if (emailAlreadyExists) {
+          throw new Error('Email already exists');
+        }
+
+        if (!verifyEmail(newEmail)) {
+          throw new Error('Invalid email');
+        }
+
+        email = newEmail;
+      }
+      if (newGender) {
+        gender = newGender;
+      }
+
+      const userUpdate = usersRepository.create({
+        username,
+        name,
+        lastName,
+        profileImageUrl,
+        bio,
+        email,
+        gender,
+      });
+
+      await usersRepository.save(userUpdate);
```

Comment:
Você não deveria chamar o update do typeorm? desta forma sua aplicação ao inves de fazer o update da entidade user, ela está sendo duplicada. Segue uma cópia da tabela na ocasião de apenas alterar executar o update:

id | username | name | email | gender | lastName | profileImageUrl | bio
-- | -- | -- | -- | -- | -- | -- | --
aac0a691-c305-40ab-8feb-1910638de0b2 | kamiokande | Bruno | brunobezerra@gmail.com | Male | Bezerra |   |  
4397ac3d-0779-4fa2-aa0a-72dd8350af10 | renatosilva | Renato | renato12@email.com | Not Specified | Souza  |   |  
77a389ad-9303-44c0-82e0-6443e9a453f0 | kamiokande | Bruno | brunobezerra@gmail.com | Male | Souza |   |  
12c2b7e6-d50c-4312-9e4b-a54f58c19777 | kamiokande | Bruno | brunobezerra@gmail.com | Male | Souza |   |  
efd280b4-4f4f-4302-baf8-e1ad3b4b730c | kamiokande | Bruno | brunobezerra@gmail.com | Male | Souza |   |  
ed3fa520-8f36-4280-b2c2-b4576d1b450d | kamiokande | Bruno | brunobezerra@gmail.com | Male | Souza |   |  